### PR TITLE
Update MSO(observability operator)

### PR DESCRIPTION
**Pull request by rewantsoni**
## Description
Update MSO(observability operator)

  1. Update the version from 0.0.8 to 0.0.10
  2. Update the name of the operator from monitoring stack to observability
  3. Install operator via OLM dependency
    

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: config/samples/mso-catalog-source-subscription.yaml [hunk 1]
```diff
@@ -1,24 +0,0 @@
...
-    operators.coreos.com/monitoring-stack-operator.openshift-operators: ''
-  name: monitoring-stack-operator
-  namespace: openshift-operators
-spec:
-  channel: development
-  installPlanApproval: Automatic
-  name: monitoring-stack-operator
-  source: monitoring-operators
-  sourceNamespace: openshift-marketplace
-  startingCSV: monitoring-stack-operator.v0.0.8
```
Comment (redhatHameed) [inline config/samples/mso-catalog-source-subscription.yaml:24]: why subscription file deleted ?

File: controllers/reconcilers/constants.go [hunk 2]
```diff
@@ -99,4 +108,14 @@ var InstallationPlatforms = map[dbaasv1alpha1.PlatformsName]dbaasv1alpha1.Platfo
...
+	dbaasv1alpha1.ObservabilityInstallation: {
+		Name:           OBSERVABILITY_NAME,
+		CSV:            OBSERVABILITY_CSV,
+		DeploymentName: OBSERVABILITY_DEPLOYMENT,
+		Image:          OBSERVABILITY_CATALOG_IMG,
+		PackageName:    OBSERVABILITY_PKG,
+		Channel:        OBSERVABILITY_CHANNEL,
+		DisplayName:    OBSERVABILITY_DISPLAYNAME,
+		Type:           dbaasv1alpha1.TypeProvider,
+	},
```
Comment (redhatHameed) [inline controllers/reconcilers/constants.go:120]: change the variable name `TypeProvider` to `TypeOperator`  WDYT ?

File: docs/install-mso.md [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
-## Steps to Remove MSO
+### Steps to Remove MSO
 
 ```commandline
 make undeploy-mso 
 // Deletes the MSO CR.
 
-make uninstall-mso
+make mso-uninstall
 // Deletes the catalog source and subscription for MSO.
```
Comment (redhatHameed) [inline docs/install-mso.md:26]: we can delete this doc and make commands, if we don't need anymore for Observability operator, and you update in readme by adding another component as `Observability operator`

File: controllers/reconcilers/constants.go (hunk 2 above)
Comment (rewantsoni) [reply controllers/reconcilers/constants.go:120]: Yes that would be much better

File: docs/install-mso.md (hunk 3 above)
Comment (rewantsoni) [reply docs/install-mso.md:26]: Yes, we can delete the doc, we will deploy the Monitoring Stack CR also via the reconciler, so we don't have to do it manually.

File: docs/install-mso.md [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
 
-## Steps to Remove MSO
+### Steps to Remove MSO
 
 ```commandline
 make undeploy-mso 
 // Deletes the MSO CR.
 
-make uninstall-mso
+make mso-uninstall
```
[the remark this answers is not in the document]
Comment (rewantsoni) [reply docs/install-mso.md]: Yes, you are right, fixed it.

File: Makefile [hunk 5]
```diff
@@ -269,12 +269,6 @@ catalog-build: opm ## Build a catalog image.
...
 	$(MAKE) docker-push IMG=$(CATALOG_IMG)
 
-mso-install:
-	oc apply -f config/samples/mso-catalog-source-subscription.yaml
-
-mso-uninstall:
-	oc delete -f config/samples/mso-catalog-source-subscription.yaml
-
 deploy-mso:
 	oc apply -f config/samples/mso.yaml
```
[the remark this answers is not in the document]
Comment (rewantsoni) [reply Makefile:273]: It's under `config/sample` of the root folder for the operator. We are going to add the creation of the monitoring Stack CR in the reconciler in the upcoming PR.

[the remark this answers is not in the document]
Comment (jmolmo) [reply Makefile:273]: @rewantsoni : Sorry, i cant find the file in `config/samples`, there is no folder `config/sample`. Could it be that the file has not been added in this PR?

[the remark this answers is not in the document]
Comment (rewantsoni) [reply Makefile:273]: https://github.com/RHEcosystemAppEng/dbaas-operator/blob/observability/config/samples/mso.yaml, I think you were looking in a different branch

[the remark this answers is not in the document]
Comment (jmolmo) [reply Makefile:273]: Now it is clear, Thanks! (I was looking for the file in the PR branch.)

**Comment by redhatHameed** [comment]
@rewantsoni is there anything remaining on this PR ?

**Comment by rewantsoni** [comment]
> @rewantsoni is there anything remaining on this PR?

No @redhatHameed We can go ahead and merge this.

File: controllers/reconcilers/constants.go (hunk 2 above)
Comment (redhatHameed) [reply controllers/reconcilers/constants.go:120]: Can we change this ?

**Pull request merged**